### ISSUE TITLE
Large image size after resize

### ISSUE THREAD

**Issue by cvbge** [bricelam/ImageResizer#100]
Hi,

I've been using IrfanView so far to resize my pictures, but I'm looking for easier to use alternative. I tried ImageResizer and from ease of use POV it's very good. But unfortunately it produces very large images compared to IrfanView:

Original image: 3.2 MB
Resized with ImageResizer: 1 **MB**
Resized with IrfanView: 115 **KB**

Both resized from 4032x3024 to 1024x768.

I compared both resized images and I do not see any difference in quality between them.

I can attach original and resized images if you want.

Would be great if ImageResizer could produce resized images with comparable small size as IrfanView. Otherwise it's not very useful for me, as I resize images to attach them on forums, so the size (in MB) matters.

**Comment by JNavas2** [comment]
Please provide the original image.

**Comment by cvbge** [comment]
Can I attach the files here it should I provide links?

**Comment by JNavas2** [comment]
@cvbge 
> Can I attach the files here it should I provide links?

Either way.

In the meantime, here's what I get resizing a 12MP 6.2 MB image
taken with my Google Pixel to 1024 x 768:
- ImageResizer 3.1 (default settings); 350 KB
- Adobe PhotoShop Elements 14 (8 High):  364 KB

![img_20180519_101520](https://user-images.githubusercontent.com/14850792/40271223-3508e2aa-5b4f-11e8-853f-bfcc265ae32d.jpg)
![img_20180519_101520 1024 x 768](https://user-images.githubusercontent.com/14850792/40271228-4242f672-5b4f-11e8-93de-0171c91357ed.jpg)
![img_20180519_101520_pse-high](https://user-images.githubusercontent.com/14850792/40271229-460840a0-5b4f-11e8-8677-d2fb69afa8f4.jpg)

**Comment by cvbge** [comment]
Attaching files:

Original:
![img_20180515_192427037](https://user-images.githubusercontent.com/3259959/40271386-f4a9defa-5b9c-11e8-8763-49de8ed40c76.jpg)

Resized with ImageResizer:
![img_20180515_192427037 sredni](https://user-images.githubusercontent.com/3259959/40271388-f7ab93fa-5b9c-11e8-8cad-e9a7165f3116.jpg)

Resized with IrfanView:
![img_20180515_192427037sm](https://user-images.githubusercontent.com/3259959/40271392-f945938c-5b9c-11e8-9110-2150f42ec5a1.jpg)

**Comment by JNavas2** [comment]
@cvbge 
When I resize your original to 1024 x 768 with Image Resizer 3.1, I get **_118 KB_**:
What version of Image Resizer and what settings are you using?
It looks like you have set Advanced > Encoding to **_90_**, which explains the size.
Default setting is **_75_**. Try that.

![original 1024 x 768](https://user-images.githubusercontent.com/14850792/40271496-f1aadb36-5b52-11e8-8f8a-ba300eb870b9.jpg)

**Comment by cvbge** [comment]
I'm using 3.1.0 version too.

I didn't change the Encoding - it's was still on 75.

I just tried setting it to 70, because I remember an issue with "75" meaning "90" in library used by ImageResizer, but even with "70" it resized to 0.99 MB.

What size do you get when you use 90?

Here are my settings:

![settings1](https://user-images.githubusercontent.com/3259959/40271977-cf0f4e74-5ba5-11e8-97c1-f03f7d25bcfc.JPG)

![settings2](https://user-images.githubusercontent.com/3259959/40271980-d28ff2c4-5ba5-11e8-9015-6f66ce5cc1b3.JPG)

Any idea what can I check more?

**Comment by cvbge** [comment]
I just tried to resize your image (the 6 MB one), and ImageResizer resized it to 203 KB (quality set to 70).
But my image is still resized to the 0.99 MB :(

**Comment by JNavas2** [comment]
Double check the size. Because of caching, Windows Explorer will still report the original size when an image is replaced.

**Comment by panther2** [comment]
I can reproduce @cbgge 's issue.
Indeed his original image - rescaled to 1024x768 at quality 70 - leads to a filesize of 0,99 MB on my Win7.64 system, too.

**Comment by panther2** [comment]
Interesting, @JNavas2 .
I reproduced what you did using Affinity Photo instead of PS, and I get 66KB when resizing, then.

**Comment by cvbge** [comment]
The image had personal data removed (in Windows 10 you can do this in Properties > Details).

Here's another image, without anything removed. This 5.55 MB image gets resized to 1.04 MB.

Both images were taken with Motorola Moto G5 Plus phone.

Note that other apps (for example IrfanView I mentioned) resize the image correctly. So it's either a problem in ImageResizer code or in library that IR uses for resizing (and I assume Irfan uses some other library).

![img_20180504_170218387](https://user-images.githubusercontent.com/3259959/40283114-9d1bf2da-5c79-11e8-9337-944d1059c75d.jpg)

**Comment by JNavas2** [comment]
@cvbge 
> The image had personal data removed

Camera make and model is personal? Regardless, you should have also made that clear at the beginning. Incomplete information makes it much harder to assess issues.

> Note that other apps (for example IrfanView I mentioned) resize the image correctly.

Not really:
* They just compress to _different sizes_, as I noted earlier, which is dependent on the JPEG encoding and on the algorithms used.
* The most likely issue is actually faulty JPEG encoding of the original. That IrfanView handles it more to your liking is a happy accident.

**Comment by cvbge** [comment]
> Camera make and model is personal? 

I didn't go into details because it does not matter. Also, according to Microsoft, apparently yes ;)

> Not really:

Let me rephrase: other apps somehow deal with the those images better.

**Comment by JNavas2** [comment]
@cvbge 

> I didn't go into details because it does not matter. 

It does matter.

> Also, according to Microsoft, apparently yes ;)

Not really.

> other apps somehow deal with the those images better.

For _your_ purposes. Not necessarily for _other_ purposes.
As I said, it's a happy accident. For you. Use what works best for you.

There are many cases where different data works differently in different apps.
That's entirely normal.

**Comment by cvbge** [comment]
> It does matter.

The problem happens also with original image, but you think it matters...

> Not really.

It''s under "personal information" but you didn't get my half-joke...

> For your purposes. Not necessarily for other purposes.

So you think having file size unnecessary big is good for some people...

**Comment by bricelam** [comment]
Image Resizer uses the built-in Windows Imaging Component (WIC) JPEG codec. Please [report](https://developercommunity.visualstudio.com/spaces/62/index.html) any issues with it to Microsoft. If you know of specific things I can do when calling the WIC APIs to produce a more optimal output, I'd love to hear them.
